ion/src/device/device.cpp

```diff
@@ -39,6 +39,22 @@ void Ion::usleep(long us) {
   }
 }
 
+extern volatile long millis_elapsed;
+
+long Ion::millis() {
+  return millis_elapsed;
+}
+
+long Ion::micros() {
+    uint32_t c1 = CM4.STCVR()->getCURRENT();
+    uint32_t ms1 = millis_elapsed;
+    uint32_t c2 = CM4.STCVR()->getCURRENT();
+    uint32_t ms2 = millis_elapsed;
+    uint32_t load = CM4.STRVR()->getRELOAD();
+
+    return ((c1 > c2) ? ms1 : ms2) * 1000 + ((load - c2) * 1000) / (load + 1);
```

Comment:
Wow, that's some non-trivial math 😄 Maybe we should split that and add a few comments?